### ISSUE TITLE
bootcoldist() should work wherever coldist() does

### ISSUE THREAD

**Issue by thomased** [rmaia/pavo#206]
`bootcoldist()` is currently RN-specific. Or mostly so anyway. It can kinda work on generic colourspaces, but it still demands `weber` arguments etc. when it shouldn't need or use them (which needs to be fixed up). It also incorrectly says it's estimating noise-weighted distances when it can't be, which I also need to fix:

```
data(sicalis)
vm <- vismodel(sicalis, relative = TRUE)
gr <- gsub("ind..", "", rownames(vm))
bootcoldist(vm, by = gr, n = c(1, 2, 2, 4), weber = 0.1, weber.achro = 0.1) 
```
```
Calculating noise-weighted Euclidean distances
```

So a few related issue to tidy up, but the bigger picture is that I'd like it to work wherever `coldist()` does (so for generic colourspaces, hexagon, etc.). If a distance can be estimated it can be bootstrapped, but this might take some refactoring as `bootcoldist()` is a bit crusty, but it's been a while since I've dived in.

**Comment by thomased** [comment]
Oh wait, some of this does work already, I just forgot to send it through `colspace()` first (🙄). Still, some doesn't — the hexagon for example.

**Comment by thomased** [comment]
Now fixed up — see #221 for the final outstanding fix.
